cmd/utxotool/utxotool.go

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2013-2016 The btcsuite developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+	"runtime"
+
+	"github.com/gcash/bchd/blockchain"
+	"github.com/gcash/bchd/blockchain/indexers"
+	"github.com/gcash/bchd/database"
+	"github.com/gcash/bchd/limits"
+	"github.com/gcash/bchlog"
+)
+
+const (
+	// blockDbNamePrefix is the prefix for the bchd block database.
+	blockDbNamePrefix = "blocks"
+)
+
+var (
+	cfg *config
+	log bchlog.Logger
+)
+
+// loadBlockDB opens the block database and returns a handle to it.
+func loadBlockDB() (database.DB, error) {
+	// The database name is based on the database type.
+	dbName := blockDbNamePrefix + "_" + cfg.DbType
+	dbPath := filepath.Join(cfg.DataDir, dbName)
+
+	log.Infof("Loading block database from '%s'", dbPath)
+	db, err := database.Open(cfg.DbType, dbPath, activeNetParams.Net)
+	if err != nil {
+		// Return the error if it's not because the database doesn't
+		// exist.
+		if dbErr, ok := err.(database.Error); !ok || dbErr.ErrorCode !=
+			database.ErrDbDoesNotExist {
+
+			return nil, err
+		}
+
+		// Create the db if it does not exist.
+		err = os.MkdirAll(cfg.DataDir, 0700)
+		if err != nil {
+			return nil, err
+		}
+		db, err = database.Create(cfg.DbType, dbPath, activeNetParams.Net)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	log.Info("Block database loaded")
+	return db, nil
+}
+
+// realMain is the real main function for the utility.  It is necessary to work
+// around the fact that deferred functions do not run when os.Exit() is called.
+func realMain() error {
+	// Load configuration and parse command line.
+	tcfg, _, err := loadConfig()
+	if err != nil {
+		return err
+	}
+	cfg = tcfg
+
+	// Setup logging.
+	backendLogger := bchlog.NewBackend(os.Stdout)
+	defer os.Stdout.Sync()
+	log = backendLogger.Logger("MAIN")
+	database.UseLogger(backendLogger.Logger("BCDB"))
+	blockchain.UseLogger(backendLogger.Logger("CHAN"))
+	indexers.UseLogger(backendLogger.Logger("INDX"))
+
+	// Load the block database.
+	db, err := loadBlockDB()
+	if err != nil {
+		log.Errorf("Failed to load database: %v", err)
+		return err
+	}
+	defer db.Close()
+
+	var utxoWriter io.Writer
+	if cfg.OutFile != "" {
+		utxoFile, err := os.Create(cfg.OutFile)
+		if err != nil {
+			log.Errorf("Unable to create file at: %s", cfg.OutFile)
+			return err
+		}
+		/*zw := gzip.NewWriter(utxoFile)
```

Comment:
The gzip compression on the output file is commented out, not sure if you have decided to drop that feature or not? Either way there is no longer a Close() on the utxoWriter/utxoFile, although as the program terminates shortly after it does not really matter, but you may want to clean that up.